pkg/signature/kms/hashivault/client.go

```diff
@@ -246,12 +246,17 @@ func (h *hashivaultClient) public() (crypto.PublicKey, error) {
 			return nil
 		},
 	)
+
+	item := h.keyCache.Get(cacheKey, ttlcache.WithLoader[string, crypto.PublicKey](loader))
 	if lerr != nil {
 		return nil, lerr
 	}
 
-	item := h.keyCache.Get(cacheKey, ttlcache.WithLoader[string, crypto.PublicKey](loader))
-	return item.Value(), lerr
+	if item == nil {
```

Comment:
Is this just to be defensive, or is there a case where an item won't be found in the cache and lerr will be nil?

Reply:
I would say, just to be more defensive, 
Because if we look at this [function](https://github.com/jellydator/ttlcache/blob/v3/cache.go#L205-L240) I see close to zero chances an item will be nil, if lerr is nil